lib/core/connection_factory/http.rb

```diff
@@ -5,7 +5,9 @@
 module Core
   module ConnectionFactory
     class Http
-      def self.build(url, timeout: 5, open_timeout: 5, retries: 2)
+      TIMEOUT = ENV.fetch('FRONTEND_HTTP_REQUEST_TIMEOUT').to_i
```

Comment:
So how about implementing a sensible default value for when the env var is missing or broken? You might be confident about environment variables in production environments always being present and correct, but I am not.